spring-shell2-jcommander-adapter/src/main/java/org/springframework/shell2/jcommander/JCommanderParameterResolver.java

```diff
@@ -74,21 +81,58 @@ public boolean supports(MethodParameter parameter) {
 
 	@Override
 	public Object resolve(MethodParameter methodParameter, List<String> words) {
+		JCommander jCommander = createJCommander(methodParameter);
+		jCommander.parse(words.toArray(new String[words.size()]));
+		return jCommander.getObjects().get(0);
+	}
+
+	private JCommander createJCommander(MethodParameter methodParameter) {
 		Object pojo = BeanUtils.instantiateClass(methodParameter.getParameterType());
-		JCommander jCommander = new JCommander();
-		jCommander.addObject(pojo);
+		JCommander jCommander = new JCommander(pojo);
 		jCommander.setAcceptUnknownOptions(true);
-		jCommander.parse(words.toArray(new String[words.size()]));
-		return pojo;
+		return jCommander;
 	}
 
 	@Override
 	public Stream<ParameterDescription> describe(MethodParameter parameter) {
```

Comment:
just realized this API was changed in a previous commit, hadn't looked into it as it wasn't part of the PR - doesn't this break the contract of a `ParameterResolver` to fit a 3rd party implementation?

from the signature:
```java
public Stream<ParameterDescription> describe(MethodParameter parameter)
```

it reads to me that a single parameter can have multiple `ParameterDescription` objects associated with it, which doesn't feel right